### ISSUE TITLE
ENC problem

### ISSUE THREAD

**Issue by JeromeSimmonds** [h-thurow/Simple-JNDI#1]
Hi,

I am trying to load:
```
InitialContext ic = new InitialContext();
ic.lookup("java:jboss/datasources/my_ds");
```
But I get the error `Invalid subcontext 'java:jboss' in context ''`

Content of the `src/test/resources/jndi.properties` file:
```
java.naming.factory.initial=org.osjava.sj.SimpleContextFactory
org.osjava.sj.root=src/test/resources/jndi
org.osjava.sj.delimiter=/
org.osjava.sj.space=java
org.osjava.sj.jndi.shared=true
```
Content of the `src/test/resources/jndi/jboss/datasources.properties` file:
```
my_ds.type=javax.sql.DataSource
my_ds.driver=com.mysql.jdbc.Driver
my_ds.url=jdbc:mysql://localhost:3306/my_ds
my_ds.user=xxx
my_ds.password=xxx
```
What is wrong? Thanks!

**Comment by h-thurow** [comment]
Hi, Jérôme,

as you configured it, the path to your datasource should be "java/jboss/datasources/my_ds”. Set 

	org.osjava.sj.space=java:jboss

Then you can refer to your datasource with "java:jboss/jboss/datasources/my_ds”.

When you would move datasources.properties to src/test/resources/jndi the path to it were "java:jboss/datasources/my_ds”. 

best regards
Holger

**Comment by JeromeSimmonds** [comment]
Hi,

Thanks for your answer.
So I changed the property in `src/test/resources/jndi.properties`:
`org.osjava.sj.space=java:jboss`

Then I moved `src/test/resources/jndi/jboss/datasources.properties` to `src/test/resources/jndi/datasources.properties`

Now
`ic.lookup("java:jboss/datasources/my_ds");`
Returns null...

**Comment by h-thurow** [comment]
Hi,

sorry for the inconveniences but congratulations, you found a bug. Other than documented when defining a datasource you can't use namespaces as "my_ds" in "my_ds.driver=...". So your datasource definition should look like

type=javax.sql.DataSource
driver=com.mysql.jdbc.Driver
url=jdbc:mysql://localhost:3306/my_ds
user=xxx
password=xxx

Name the file my_ds.properties. Make a directory "src/test/resources/jndi/datasources/" and move the file to it.

**Comment by JeromeSimmonds** [comment]
Thanks!

**Comment by h-thurow** [comment]
It is a bit unusual to use "/" in property names, and I am not happy with
it but it is the way it is implemented. But I am working on a way to
combine dots in configuration files with slashes in lookup pathes. So
revisit this site to get the next update.

**Comment by h-thurow** [comment]
Today I released 0.14.0. You can now use slash separated lookup pathes with dot separated property names. See https://github.com/h-thurow/Simple-JNDI/wiki/Use-slash-separated-lookup-pathes-with-dot-separated-property-names-(New-in-0.14.0) . 
Also new: https://github.com/h-thurow/Simple-JNDI/wiki/A-more-elegant-way-to-lookup-typed-properties-(New-in-0.14.0)
